lib/components/base_component.dart

```diff
@@ -0,0 +1,153 @@
+import 'dart:ui';
+
+import 'package:meta/meta.dart';
+import 'package:ordered_set/comparing.dart';
+import 'package:ordered_set/ordered_set.dart';
+
+import '../effects/effects.dart';
+import '../effects/effects_handler.dart';
+import '../extensions/vector2.dart';
+import '../game.dart';
+import '../text_config.dart';
+import 'component.dart';
+
+/// This can be extended to represent a basic Component for your game.
+///
+/// The difference between this and [Component] is that the [BaseComponent] can
+/// have children, handle effects and can be used to see whether a position on
+/// the screen is on your component, which is useful for handling gestures.
+abstract class BaseComponent extends Component {
+  final EffectsHandler _effectsHandler = EffectsHandler();
+
+  final OrderedSet<Component> _children =
```

Comment:
should this be a list of BaseComponent? does it make sense to have children that can't benefit from the inheritance?

Reply:
Yes, they can be leaf nodes even though they can't have children themselves.

Reply:
but what do they gain? like if the don't have position, dont receive events, etc, why would you add them as a child? I am ok with it just posing a philosophical question to what makes more sense

Reply:
The children doesn't have to have anything to do with positioning anymore, it could be anything that you want to have separated by concern. Say that you have a timer that is a child of a component for example and you want to stop that timer when a component is clicked, then that click could be propagated to the timer. Quite bad example, but maybe you get what I mean?